wcomponents-core/src/main/java/com/github/bordertech/wcomponents/WFileWidget.java

```diff
@@ -110,12 +133,104 @@ protected boolean doHandleRequest(final Request request) {
 		boolean changed = value != null || current != null;
 
 		if (changed) {
-			setData(value);
-		}
+			// Reset validation fields
+			resetValidationState();
+			// if User Model exists it will be returned, othewise Shared Model is returned
+			final FileWidgetModel sharedModel = getComponentModel();
+			// if fileType is supplied then validate it
+			if (hasFileTypes()) {
+				boolean validFileType = FileUtil.validateFileType(value, getFileTypes());
+				// If invalid only then update 
+				if (sharedModel.validFileType != validFileType) {
+					// if User Model exists it will be returned, othewise it will be created
+					final FileWidgetModel userModel = getOrCreateComponentModel();
+					userModel.validFileType = validFileType;
+				}
+			}
+			
+			// if fileSize is supplied then validate it
+			if (hasMaxFileSize()) {
+				boolean validFileSize = FileUtil.validateFileSize(value, getMaxFileSize());
+				// If invalid only then update 
+				if (sharedModel.validFileSize != validFileSize) {
+					// if User Model exists it will be returned, othewise it will be created
+					final FileWidgetModel userModel = getOrCreateComponentModel();
+					userModel.validFileSize = validFileSize;
+				}
+			}
+			
+			// if file is valid, the update data
+			if (isFileSizeValid() && isFileTypeValid()) {
+				setData(value);
+			} else if (current == null) {
+				// otherwise no change
+				changed = false;
+			} else {
+				changed = true;
+				setData(null);
+			}
+		} 
 
 		return changed;
 	}
 
+	/**
+	 * Reset validation state.
+	 */
+	private void resetValidationState() {
+		// if User Model exists it will be returned, othewise Shared Model is returned
+		final FileWidgetModel componentModel = getComponentModel();
+		// If Shared Model is returned then both fileType and fileSize are always valid
+		// If User Model is returned then reset back to true
+		if (!componentModel.validFileSize || !componentModel.validFileType) {
+		    // If the state has changed then we know for certain a user model exists
```

Comment:
getOrCreateComponentModel before update.

Reply:
```java
final FileWidgetModel componentModel = getComponentModel();
```
If `componentModel` is a **Shared Model** then both `fileType` and `fileSize` is always *True*, then `if` fails.
If `componentModel` is a **User Model** then both `fileType` and `fileSize` can be modified as desired
Do you want to still create another variable inside `if` as below?
```java
final FileWidgetModel userModel = getOrCreateComponentModel();
userModel.validFileType = true;
userModel.validFileSize = true;
```
I added a new unit test `testValidationMultiUIContext` to test valid and invalid file